crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -198,48 +193,37 @@ impl Persistence {
             .pool
             .begin()
             .await
-            .context("begin")
-            .map_err(error::Auction::DbError)?;
+            .map_err(error::Auction::BadCommunication)?;
 
-        let deadline = database::settlement_scores::fetch(&mut ex, auction_id)
-            .await
-            .context("fetch scores")
-            .map_err(error::Auction::DbError)?
-            .ok_or(error::Auction::Missing)
-            .map(|scores| (scores.block_deadline as u64).into())?;
+        let surplus_capturing_jit_order_owners =
+            database::surplus_capturing_jit_order_owners::fetch(&mut ex, auction_id)
+                .await
+                .map_err(error::Auction::BadCommunication)?
+                .ok_or(error::Auction::NotFound)?
+                .into_iter()
+                .map(|owner| eth::H160(owner.0).into())
+                .collect();
 
         let prices = database::auction_prices::fetch(&mut ex, auction_id)
             .await
-            .context("fetch auction prices")
-            .map_err(error::Auction::DbError)?
+            .map_err(error::Auction::BadCommunication)?
             .into_iter()
             .map(|price| {
                 let token = eth::H160(price.token.0).into();
                 let price = big_decimal_to_u256(&price.price)
                     .ok_or(domain::auction::InvalidPrice)
                     .and_then(|p| domain::auction::Price::new(p.into()))
-                    .map_err(error::Auction::Price);
+                    .map_err(|_err| error::Auction::InvalidPrice(token));
```

Comment:
Maybe worth chaining the error into `InvalidPrice` too.

Reply:
Hm, not sure worth it, i think printing token should be enough to reconstruct the problem.